akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/Fog.scala

```diff
@@ -0,0 +1,84 @@
+package sample.killrweather.fog
+
+import java.net.{ DatagramSocket, InetSocketAddress }
+import java.nio.channels.DatagramChannel
+
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.ActorSystem
+import com.typesafe.config.{ Config, ConfigFactory }
+
+/**
+  * In another terminal start the `Fog` (see Fog computing https://en.wikipedia.org/wiki/Fog_computing).
+  * Starts the fog network, simulating devices and stations.
+  * In the wild, each station would run its own system and be location-aware.
+  */
+object Fog {
+
+  def main(args: Array[String]): Unit = {
+    val ports = if (args.isEmpty) Seq(8081) else args.map(_.toInt).toSeq
+    val weatherPorts = ports.flatMap(findHttpPort).toVector
+
+    val settings = FogSettings()
+
+    // simple sample, using one node on local machine
+    val rootBehavior = Behaviors.setup[Nothing] { context =>
+      (1 until settings.weatherStations).foreach { n =>
+        val wsid = WeatherStation.WmoId(n.toString)
+        val weatherPort = weatherPorts(n % weatherPorts.size)
+
+        context.spawn(WeatherStation(wsid, settings, weatherPort), s"weather-station-${wsid.id}")
+
+        // for sample reasons only
+        Thread.sleep(settings.staggerStartup.toMillis)
```

Comment:
um, looking in other direction

Reply:
Ya, this is the last thing to do. It just needs a breather to not overload the connections for all 50 starting up at the same time.